template-processors/base/bin/resourceManager.sh

```diff
@@ -54,6 +54,41 @@ function addLabels() {
     done
 }
 
+# appendResourceVersion - patches the YAML&JSON files in $MANIFEST_DIR,
+# adding the metadata.resourceVersion for each resource being managed.
+# This is intended to serve as a locking mechanism when applying resources
+# in which Kubernetes will fail the apply with a StatusConflict (HTTP status code 409)
+# Ref https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api-conventions.md#concurrency-control-and-consistency
+function appendResourceVersion() {
+    local tmpdir="$(mktemp -d)"
+    mkdir -p "$tmpdir/$MANIFEST_DIR"
+    mkdir -p "$tmpdir/processed_files/$MANIFEST_DIR"
+    # shellcheck disable=SC2044
+    for file in $(find "$MANIFEST_DIR" -regextype posix-extended -iregex '.*\.(ya?ml|json)'); do
+        mkdir -p "$tmpdir"/"$(dirname "$file")"
+        mkdir -p "$tmpdir"/processed_files/"$(dirname "$file")"
+        kube get --ignore-not-found -f "$file" -o yaml >"$tmpdir/$file"
+        local kind="$(cat "$tmpdir"/"$file" | yq -y .kind)"
+        if [[ "$kind" =~ "List" ]]; then
+            #TODO: figure out how to handle this
```

Comment:
@vinny-sabatini how were you able to trigger a `kind: List` being returned from the k8s API server? Did the list only container a single element or did it contain multiple elements?


Looking at the other code that we have in this script. We did not currently support multiple YAML objects in a single  file. So if the API server returns list with more than one element than I'm not sure how we would handle that.

My naive solution would be to add something this ....
```
if [ number_of_list_elements != 1 ]; then
    # something is wrong
    exit 1
fi
# now process the first list element
```

Reply:
@sanbornick and @seegeekrow feel free to add your thoughts.

Reply:
I achieved this by retrieving the `hello-world-yaml` resources after I deployed them to the cluster.
```
$ kubectl get -f eunomia/examples/hello-world-yaml/template1/hello-world.yaml
```

Reply:
This example is causing a `List` to be returned.
https://github.com/KohlsTechnology/eunomia/blob/master/examples/hello-world-yaml/template1/hello-world.yaml

Reply:
[This](https://github.com/KohlsTechnology/eunomia/blob/754565064f351610598cd485cc089867377d8cd4/template-processors/base/bin/resourceManager.sh#L47-L54) I assume is applying the annotations to just the first document, which also is a gap in testing as we should check for unintentionally unmanaged objects, for  [this example](https://github.com/KohlsTechnology/eunomia/blob/master/examples/hello-world-yaml/template1/hello-world.yaml). I would just stub it like that for now and open a new 🐛. This needs to support multiple docs as that pattern is ubiquitous in the community.

Reply:
wrote that a day ago, not sure why it just posted now...